Automattic/woocommerce-payments
# Terminal locations: harden address data validation

Fixes #3071

#### Changes proposed in this Pull Request

- Hardened address data validation as requested
- TBD

#### Testing instructions

- Modify `includes/admin/class-wc-rest-payments-terminal-locations-controller.php` file and replace `'permission_callback' => [ $this, 'check_permission' ]` with `'permission_callback' => '__return_true'` for the `$this->rest_base . '/store'` route.
- Ensure the address is properly populated in WC settings
- Execute `curl -X GET {store_url}/wp-json/wc/v3/payments/terminal/locations/store`
- Observer the location address as a response
- Update WC setting so address city/line1/zip is empty
- Execute `curl -X GET {store_url}/wp-json/wc/v3/payments/terminal/locations/store`
- Observer the error as a response

-------------------
